### ISSUE TITLE
td-has-header should not always fail <td headers="">

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2039]
The `td-has-header` rule seems to consistently fail `<td headers="">` elements. We have some test data that shows that AT will fall back to column / rowheaders in this situation.

```html
<table>
  <tr>
    <th scope="col">Projects</th>
    <th scope="col">Progress</th>
  </tr>
  <tr>
    <td headers="">My Project</td>
    <td>15%</td>
  </tr>
</table>
```

**IE + JAWS**: Works
**Chrome + JAWS**: Works
**Firefox + NVDA**: Works
**Safari + VO**: Works

I think we should report this for review. Axe-core should probably still fail if the headers attribute references something that doesn't exist, or something outside the table.

**Comment by jkodu** [comment]
@WilcoFiers did you mean the rule `td-headers-attr` here?

**Comment by WilcoFiers** [comment]
Nope. The problem here is that td-has-header stops looking if it finds a headers attribute without associated headers. It should continue looking for column/rowheaders. This is not according to spec, but we have test data showing this happens consistently.

**Comment by padmavemulapati** [comment]
Its verified as part of the verification of `https://github.com/dequelabs/axe-core/issues/2013` , yes we are seeing the results as `incomplete` which is expected as per the change done here in this ticket.
![image (11)](https://user-images.githubusercontent.com/47409579/77031885-ebe12d00-69c8-11ea-8918-250f22f0380f.png)

**Comment by straker** [comment]
No docs needed
